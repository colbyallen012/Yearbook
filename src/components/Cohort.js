import React from 'react';
import Person from './Person';
import Student from './Student';
import './Cohort.css';

const Cohort = ({staff, students}) => {

  const staffCards = staff.map(membs => {
    return (
      <Person 
        photo={membs.photo}
        name={membs.name}
        quote={membs.quote}
        superlative={membs.superlative}
        key={membs.id}
      />
    )
  })

  const studentCards = students.map(studs => {
    return (
      <Student 
        photo={studs.photo}
        name={studs.name}
        quote={studs.quote}
        superlative={studs.superlative}
        key={studs.id}
      />
    )
  })

  return (
    <main>
      <div className='staff-container'>
        {staffCards}
      </div>
      <h2>Students</h2>
      <div className='student-container'>
        {studentCards}
      </div>
    </main>
  )
}

export default Cohort;
