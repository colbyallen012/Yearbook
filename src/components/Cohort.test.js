import React from 'react';
import { shallow, mount } from 'enzyme';
import Cohort from './Cohort.js';

const mockStaff = [
  {id: 1,
  name:'Leta', 
  quote:'Take a ballet pom', 
  superlative:'most likely to learn me React'}
]

const mockStudents = [
  {id: 1,
  name:'Colby', 
  quote:'Im hungry', 
  superlative:'most likely to win an eating competition'}
]

describe('Cohort', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Cohort 
        staff={mockStaff}
        students={mockStudents}
      />
    )
  })
  it('should match the snapshot with all the data passed in correctly', () => {
    expect(wrapper).toMatchSnapshot()
  });
});