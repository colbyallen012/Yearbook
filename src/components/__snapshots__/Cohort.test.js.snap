// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cohort should match the snapshot with all the data passed in correctly 1`] = `
<main>
  <div
    className="staff-container"
  >
    <Person
      key="1"
      name="Leta"
      quote="Take a ballet pom"
      superlative="most likely to learn me React"
    />
  </div>
  <h2>
    Students
  </h2>
  <div
    className="student-container"
  >
    <Student
      key="1"
      name="Colby"
      quote="Im hungry"
      superlative="most likely to win an eating competition"
    />
  </div>
</main>
`;
